ProcessList.c

```diff
@@ -553,13 +554,39 @@ void ProcessList_rebuildPanel(ProcessList* this) {
    bool foundFollowed = false;
 
    for (int i = 0; i < processCount; i++) {
-      Process* p = (Process*) Vector_get(this->processes, i);
-
-      if ( (!p->show)
-         || (this->userId != (uid_t) -1 && (p->st_uid != this->userId))
-         || (incFilter && !(String_contains_i(Process_getCommand(p), incFilter)))
-         || (this->pidMatchList && !Hashtable_get(this->pidMatchList, p->tgid)) )
-         continue;
+       Process* p = (Process*) Vector_get(this->processes, i);
+
+       if ( (!p->show)
+               || (this->userId != (uid_t) -1 && (p->st_uid != this->userId))
+               || (incFilter && !(String_contains_i(Process_getCommand(p), incFilter))) ) {
+           continue;
+       } else {
+           if (this->pidMatchList && !Hashtable_get(this->pidMatchList, p->tgid)) {
+               // iterate over parents and check hashtable for entries
+               pid_t ppid = Process_getParentPid(p);
+               bool preExisting;
+               bool nextPreExisting;
+               Process *ppidProcess = ProcessList_getProcess(this, ppid, &preExisting, LinuxProcess_new);
+               Process *nextProcess;
```

Comment:
No reason to fetch processes from the platform API as there's a perfectly fine list of proceses currently tracked just part of the `ProcessList` instance you are operating on.

Reply:
I updated it to find processes from the process table.